File: Java/src/DynamicArray/TestDynamicArray.java
package DynamicArray;

public class TestDynamicArray {
    public static void main(String[] args) {
        DynamicArray<Integer> dynamicArray = new DynamicArray<Integer>();

        for (int i = 1; i < 10; i++) {
            dynamicArray.add(i * 2);
        }


        System.out.println(dynamicArray.toString());
        System.out.println("Size: " + dynamicArray.size());
        System.out.println("Capacity: " + dynamicArray.capacity());
        System.out.println("Index: " + dynamicArray.indexOf(8));
        System.out.println("Contain: " + dynamicArray.contain(200));


        System.out.println("Get: " + dynamicArray.get(3));
        dynamicArray.set(3, 20);
        System.out.println("Set: " + dynamicArray.toString());


        dynamicArray.removeAt(3);
        System.out.println("Remove at: " + dynamicArray.toString());

        dynamicArray.remove(6);
        System.out.println("Remove: " + dynamicArray.toString());

        dynamicArray.clear();
        System.out.println("Clear: " + dynamicArray.toString());

    }
}

File: Typescript/DynamicArray/TestDynamicArray.ts
import {DynamicArray} from "./DynamicArray";

let array: DynamicArray<number> = new DynamicArray<number>();

console.log(`isEmpty: ${array.isEmpty()}`)

for (let i = 0; i < 19; i++) {
    array.add(i * 2)
}

console.log(`Capacity: ${array.volume()}`)
console.log(`Size: ${array.length()}`)
console.log(`isEmpty: ${array.isEmpty()}`)

console.log(`getAt: ${array.getAt(4)}`)
console.log(`setAt: ${array.setAt(4, 100)}`)
console.log(`indexOf: ${array.indexOf(100)}`)
console.log(`indexOf: ${array.indexOf(8)}`)
console.log(`Contains true: ${array.contains(100)}`)
console.log(`Contains false: ${array.contains(2808)}`)
console.log(`remove: ${array.removeAt(1)}`)
console.log(`remove: ${array.remove(6)}`)
console.log(`toString After remove: ${array.toString()}`);



console.log(`Capacity: ${array.volume()}`)
console.log(`Size: ${array.length()}`)
console.log(`toString: ${array.toString()}`);

array.clear();
console.log(`toString After Clear: ${array.toString()}`);
